includes/class-receiver.php

```diff
@@ -351,9 +351,19 @@ public static function post( $request ) {
 				$commentdata['comment_meta'] = array();
 			}
 
-			// save comment
+			// save comment but remove content filtering because we filter our own content
+			remove_filter( 'pre_comment_content', 'wp_filter_post_kses' );
+			remove_filter( 'pre_comment_content', 'wp_filter_kses' );
+
 			$commentdata['comment_ID'] = wp_new_comment( $commentdata, true );
 
+			// restore filter after add
+			if ( current_user_can( 'unfiltered_html' ) ) {
```

Comment:
This is an API request, so there is no current_user!?!

Reply:
what if there is no current_user? will it fail gracefully?

Reply:
Yes.